### ISSUE TITLE
Docker 1.6 + Compose 1.2 : Unable to bind mount a file if it already exists in the image

### ISSUE THREAD

**Issue by frntn** [docker/compose#1326]
### What I do

Trying to create a container with a bind mount on a file existing in the image.
I use `docker-compose -f lb.yaml up -d` with the following `lb.yaml` 

```yaml
loadbalancer:
  ports:
    - 443:443
  volumes:
    - /tmp/haproxy.cfg:/etc/haproxy/haproxy.cfg
  image: ekino/haproxy:base
```

### What I get

Every time, even from a clean install (no image / no container yet), I get something like that :
```
500 Internal Server Error: file exists at /var/lib/docker/aufs/mnt/d6b22d7cc86c3798687cc06ef7923557c076dddf4ecbd6810bd460bed20a94ec/etc/haproxy/haproxy.cfg, can't create volume there
```

### What I expected

I expected the same behaviour as with the equivalent cli command which is **always OK**
```bash
docker run -d \
    --name loadbalancer \
    -p 443:443 \
    -v /tmp/haproxy.cfg:/etc/haproxy/haproxy.cfg \
    ekino/haproxy:base
```

**Comment by frntn** [comment]
I have tcpdumped the calls and I can see the HTTP requests urls are both the same :
```
# cli
POST /v1.15/containers/create?name=loadbalancer HTTP/1.1
# compose
POST /v1.15/containers/create?name=loadbalancer_1 HTTP/1.1
```

But the data posted along with the requests are not as you can see in this diff :
https://gist.github.com/frntn/c60fc68f0de8f8fd7d62/revisions

Besides other oddities we can see : 
*  `compose` is using the **Volumes** json parameter which end up with a **500 Internal Server Error** [since docker 1.6](https://github.com/docker/docker/pull/10112/files),
* whereas `cli` is using the **Binds** json parameter which end up with the expected **201 Created**

**Comment by rickystillwell** [comment]
I'm also hitting this error after upgrading from Docker 1.5 to 1.6

file exists at /data/lib/docker/aufs/mnt/cc379d2061cd295eb0ca6ef990799dcbcc06770d510a6b4c9e89ff25dc5e9926/opt/jetty-distribution-9.2.6.v20141205/start.ini, can't create volume there

**Comment by intellectronica** [comment]
Same problem after upgrading to 1.6 (using fig):
`file exists at /mnt/sda1/var/lib/docker/aufs/mnt/fa1aa24d9bdb47f058487c031de133ad927855249be73c054f2afba1e4abfc0d/app/nginx.tmpl, can't create volume there`

**Comment by ckortekaas** [comment]
I had this as well with docker 1.6 until I realised I was still on docker-compose 1.1. 

When I upgraded docker-compose to 1.2 (http://docs.docker.com/compose/install/) it all started working ok with docker 1.6. But maybe I have some strange edge-case.

**Comment by dnephin** [comment]
It could have been fixed by https://github.com/docker/compose/commit/eef4bc39175913e82eb8d814f3f4b03df1f04f16 which was new in 1.2

**Comment by yomguy** [comment]
Fixed also updating docker-compose to 1.2

**Comment by aanand** [comment]
@frntn What version of Compose are you using?

**Comment by frntn** [comment]
Ok I now understand. 

I have updated this issue title after checking my docker-compose at system level (default `PATH`), but my jenkins jobs use a local docker-compose located in their workspace.
**The workspaces hadn't been wiped out and were still in 1.1**

I cannot not update these workspaces right now but seems legit to close the issue as multiple users confirm it is fixed with 1.2.
Thanks everyone for your feedbacks

**Comment by wardi** [comment]
I'm running in to this trying to use the API from docker-py. Is there a docker-py version I need to use for this API change? I'll try working around the issue by requesting an older api version.
